### PR TITLE
DOC: suggestions for `scipy.sparse` release notes

### PR DESCRIPTION
**Pull request by jjerphan**
#### Reference issue

Suggestions for https://github.com/scipy/scipy/pull/18671.

#### What does this implement/fix?

A few suggestions for presenting the recent work on the `scipy.sparse` module.

#### Additional information

Thank you for managing the release of 1.11, @tylerjereddy!

### CODE REVIEW

File: doc/source/release/1.11.0-notes.rst [hunk 1]
```diff
@@ -106,23 +107,23 @@ New features
 
 `scipy.sparse` improvements
 ===========================
-- `scipy.sparse` array (not matrix) classes now return a sparse array instead
-  of a dense array when divided by a dense array.
-- A new public base class `scipy.sparse.sparray` was introduced, allowing
+- A new public base class `scipy.sparse.sparray` was introduced, allowing further
+  extension of the sparse array API (like support for 1- and n-dimensional sparse
+  arrays) without breaking backwards compatibility.
```
Comment (jjerphan) [inline doc/source/release/1.11.0-notes.rst]:
```suggestion
  extension of the sparse array API (such as the support for 1- and
  n-dimensional sparse arrays) without breaking backwards compatibility.
```

File: doc/source/release/1.11.0-notes.rst [hunk 2]
```diff
@@ -106,23 +107,23 @@ New features
...
   while `isinstance(x, scipy.sparse.spmatrix)` selects only the old sparse
-  matrix types.
-- The behavior of `scipy.sparse.isspmatrix()` was updated to return True for
-  only the sparse matrix types. If you want to check for either sparse arrays
-  or sparse matrices, use `scipy.sparse.issparse()` instead. (Previously,
-  these had identical behavior.)
-- Sparse arrays constructed with 64-bit indices will no longer automatically
-  downcast to 32-bit.
+  matrix classes.
+- Division of sparse arrays by a dense array now return sparse arrays
```
Comment (jjerphan) [inline doc/source/release/1.11.0-notes.rst]:
```suggestion
- Division of sparse arrays by a dense array now return sparse arrays.
```

File: doc/source/release/1.11.0-notes.rst [hunk 3]
```diff
@@ -106,23 +107,23 @@ New features
 
 `scipy.sparse` improvements
 ===========================
-- `scipy.sparse` array (not matrix) classes now return a sparse array instead
-  of a dense array when divided by a dense array.
-- A new public base class `scipy.sparse.sparray` was introduced, allowing
+- A new public base class `scipy.sparse.sparray` was introduced, allowing further
+  extension of the sparse array API (such as the support for 1- and
+  n-dimensional sparse arrays) without breaking backwards compatibility.
```
Comment (jjerphan) [inline doc/source/release/1.11.0-notes.rst]:
```suggestion
  extension of the sparse array API (such as the support for 1-dimensional
  sparse arrays) without breaking backwards compatibility.
```

File: doc/source/release/1.11.0-notes.rst [hunk 4]
```diff
@@ -106,23 +107,23 @@ New features
...
+  `scipy.sparse.issparse` now has to be used instead to check for instances of sparse
+  arrays or instances of sparse matrices.
+- Sparse arrays constructed with int64 indices will no longer automatically
+  downcast to int32.
 - A new `scipy.sparse.diags_array` function was added, which behaves like the
   existing `scipy.sparse.diags` function except that it returns a sparse
   array instead of a sparse matrix.
-- ``argmin`` and ``argmax`` methods now return the correct result when no
-  implicit zeros are present.
+- The `argmin` and `argmax` methods now return the correct result when explicit
```
Comment (tylerjereddy) [inline doc/source/release/1.11.0-notes.rst]:
```suggestion
- The ``argmin`` and ``argmax`` methods now return the correct result when explicit
```

The double backticks are used in our release notes for any "objects" that aren't fully resolved in the SciPy namespace, so I'll restore those.

File: doc/source/release/1.11.0-notes.rst [hunk 5]
```diff
@@ -106,23 +107,23 @@ New features
...
-  matrix types.
-- The behavior of `scipy.sparse.isspmatrix()` was updated to return True for
-  only the sparse matrix types. If you want to check for either sparse arrays
-  or sparse matrices, use `scipy.sparse.issparse()` instead. (Previously,
-  these had identical behavior.)
-- Sparse arrays constructed with 64-bit indices will no longer automatically
-  downcast to 32-bit.
+  matrix classes.
+- Division of sparse arrays by a dense array now return sparse arrays.
+- `scipy.sparse.isspmatrix` now only return `True` for the sparse matrices instances.
```
Comment (tylerjereddy) [inline doc/source/release/1.11.0-notes.rst]:
```suggestion
- `scipy.sparse.isspmatrix` now only returns `True` for the sparse matrices instances.
```

File: doc/source/release/1.11.0-notes.rst [hunk 6]
```diff
@@ -106,23 +107,23 @@ New features
...
   while `isinstance(x, scipy.sparse.spmatrix)` selects only the old sparse
-  matrix types.
-- The behavior of `scipy.sparse.isspmatrix()` was updated to return True for
-  only the sparse matrix types. If you want to check for either sparse arrays
-  or sparse matrices, use `scipy.sparse.issparse()` instead. (Previously,
-  these had identical behavior.)
-- Sparse arrays constructed with 64-bit indices will no longer automatically
-  downcast to 32-bit.
+  matrix classes.
+- Division of sparse arrays by a dense array now return sparse arrays.
```
Comment (tylerjereddy) [inline doc/source/release/1.11.0-notes.rst]:
```suggestion
- Division of sparse arrays by a dense array now returns sparse arrays.
```

**Review by tylerjereddy — COMMENTED** [review]
Looks good, and the docs build seems "ok" for me locally on this branch. I made a few inline suggestions I'll apply through the GitHub interface then merge it.

**Pull request merged**